src/main/java/edu/wpi/grip/ui/SocketControlView.java

```diff
@@ -0,0 +1,172 @@
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A JavaFX control that renders a {@link Socket}.  This control shows a label with the socket's identifier, as well as
+ * an optional control such as a slider or a number spinner that lets the user interact with the value in the socket.
+ * <p>
+ * This class both subscribes to and sends {@link SocketChangedEvent}s on the event bus, and acts as an intermediary
+ * between EventBus events and a JavaFX property.
+ */
+public class SocketControlView extends AnchorPane implements Initializable {
+    @FXML
+    private Label identifier;
+
+    @FXML
+    private StackPane controlPane;
+
+    private final EventBus eventBus;
+    private final Socket socket;
+    private final Property valueProperty;
+
+    public SocketControlView(EventBus eventBus, Socket<?> socket) {
+        checkNotNull(eventBus);
+        checkNotNull(socket);
+
+        this.eventBus = eventBus;
+        this.socket = socket;
+        this.valueProperty = new SimpleObjectProperty();
+
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("SocketControl.fxml"));
+            fxmlLoader.setRoot(this);
+            fxmlLoader.setController(this);
+            fxmlLoader.load();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Always set the socket to the value of valueProperty, which may be bound to a GUI property.
+        this.valueProperty.addListener((observableValue, o, t1) -> this.socket.setValue(observableValue.getValue()));
+
+        this.eventBus.register(this);
+    }
+
+    @Override
+    public void initialize(URL url, ResourceBundle resourceBundle) {
+        SocketHint<?> socketHint = this.socket.getSocketHint();
+        Object[] domain = socketHint.getDomain();
+
+        // Set the label on the control based on the identifier from the socket hint
+        this.identifier.setText(socketHint.getIdentifier());
+
+        // Set an input control of some sort based on the "view" field of the socket hint
+        switch (socketHint.getView()) {
+            case NONE:
+                // The hint says not to show a control, so there's nothing to do here.
+                break;
+
+            case SPINNER: {
+                // To use a spinner, the socket should be able to hold a number
+                checkArgument(Number.class.isAssignableFrom(socketHint.getType()));
+
+                SpinnerValueFactory spinnerValueFactory;
+                Number initialValue = (Number) socket.getValue();
+
+                Number min = Double.MIN_VALUE, max = Double.MAX_VALUE;
+                if (domain != null && domain.length == 2) {
+                    // A spinner can have a min and max, if the domain holds two values.
+                    min = (Number) domain[0];
+                    max = (Number) domain[1];
+                }
```

Comment:
The idea with spinners is that the domain is optional, since there isn't really a need for a spinner to have an upper and lower bound (other than those imposed by the min/max values a double can hold)

I guess we could be really pedantic and throw an error if the domain exists but isn't something we can use, but I kind of want to avoid throwing errors for things relating to the hint.  The name itself implies that the world won't end if the current implementation can do exactly what it specified.

Once we have logging and error reporting in place (which we should eventually have) it might be a good idea to put a log statement anywhere that a hint is not 100% respected by the GUI.